src/ucp/core/ucp_am.h

```diff
@@ -28,6 +28,15 @@ enum {
 };
 
 
+enum {
+    UCP_AM_SEND_PRIV_FLAG_FIRST_FLAG  = UCS_BIT(15),
+
+    /* Indicates that user header was copied 
+     * to internal UCP buf and need to be freed. */
+    UCP_AM_SEND_PRIV_FLAG_FREE_HEADER = UCP_AM_SEND_PRIV_FLAG_FIRST_FLAG
```

Comment:
Is it possible to set req->send.msg_proto.am.header to `NULL` when we do not really need to free it?
In this case we will not need this flag

Reply:
Hi
This field initially store the user header buffer passed to ucp_am_send_nbx.
How you differ between buffer allocated by the user and buffer allocated by UCX?

Reply:
can use lowest bit of pointer, for example